### ISSUE TITLE
Custom Forbidden error

### ISSUE THREAD

**Issue by asdf404** [yanickrochon/koa-rbac#8]
Is it possible to throw custom error on failed access check? Now `koa-rbac` sets `ctx.status` to 403 and `ctx.body` to `"Forbidden"`. But it is not usable in my API, because clients expect a JSON and backend always should response with JSON, like: `{ "error": "E_ACCESS", "message": "You are not allowed to perform this action" }`.
If I manually change [here](https://github.com/yanickrochon/koa-rbac/blob/b118b5f7c1528f99d6de259307790e74bbeffe36/rbac.js#L59-L68) to:

```js
function restrict(redirectUrl) {
  if (redirectUrl) {
    return ctx.redirect(redirectUrl);
  } else {
    // here we throw HTTP error, instead of change `ctx.status`
    throw new NotPermittedError()
  }
}
```

it works fine and my error-handler can catch throwed errors and handle.
I'm using `koa@2` and `koa-router@7` if it matters.

**Comment by yanickrochon** [comment]
Alright, I will add an option for this, then; an `errorHandler` receiving three args : `ctx`, `next` and `redirectUrl`.

```js
function restrict(redirectUrl) {
  if (redirectUrl) {
    return ctx.redirect(redirectUrl);
  } else if (options.errorHandler) {
    return options.errorHandler(ctx, next, redirectUrl);
  } else {
    ctx.status = 403;
    ctx.body = 'Forbidden';
  }
}
```

How's that?

**Comment by asdf404** [comment]
That's will be awesome! Thanks.

**Comment by yanickrochon** [comment]
The `restrictionHandler` is a function receiving 3 arguments :+1: 

* `ctx` : the request context
* `permissions` : the actual permissions passed to the `allow` / `deny` / `check` middlewares.
* `redirectUrl` : if specified, the value of the redirection

**Comment by borodyadka** [comment]
Can you publish this to NPM or tag this revision on Github?

**Comment by yanickrochon** [comment]
npm published as `2.0.1`!

**Comment by borodyadka** [comment]
Thanks
